sqlmesh/cli/project_init.py

```diff
@@ -0,0 +1,508 @@
+                    required_fields.append(option_str)
+                else:
+                    non_required_fields.append(option_str)
+
+            connection_settings = "".join(required_fields + non_required_fields)
+
+            engine_link = f"/{engine_type}/#connection-options"
+
+        connection_settings = (
+            "      # For more information on configuring the connection to your execution engine, visit:\n"
+            "      # https://sqlmesh.readthedocs.io/en/stable/reference/configuration/#connection\n"
+            f"      # {doc_link.format(engine_link=engine_link)}\n{connection_settings}"
+        )
+
+    default_configs = {
+        ProjectTemplate.DEFAULT: f"""# --- Gateway Connection ---
+gateways:
+  {engine_type}:
+    connection:
+{connection_settings}
+default_gateway: {engine_type}
+
+# --- Model Defaults ---
+# https://sqlmesh.readthedocs.io/en/stable/reference/model_configuration/#model-defaults
+
+model_defaults:
+  dialect: {DIALECT_TO_TYPE.get(engine_type)}
+  start: {start or yesterday_ds()} # Start date for backfill history
+  cron: '@daily'    # Run models daily at 12am UTC (can override per model)
+
+# --- Linting Rules ---
+# Enforce standards for your team
+# https://sqlmesh.readthedocs.io/en/stable/guides/linter/
+
+linter:
+  enabled: true
+  rules:
+    - ambiguousorinvalidcolumn
+    - invalidselectstarexpansion
+""",
+        ProjectTemplate.DBT: """from pathlib import Path
+
+from sqlmesh.dbt.loader import sqlmesh_config
+
+config = sqlmesh_config(Path(__file__).parent)
+""",
+    }
+
+    default_configs[ProjectTemplate.EMPTY] = default_configs[ProjectTemplate.DEFAULT]
+    default_configs[ProjectTemplate.DLT] = default_configs[ProjectTemplate.DEFAULT]
+
+    flow_cli_mode = """
+# FLOW: Minimal prompts, automatic changes, summary output
+# https://sqlmesh.readthedocs.io/en/stable/reference/configuration/#plan
+
+plan:
+  no_diff: true             # Hide detailed text differences for changed models
+  no_prompts: true          # No interactive prompts
+  auto_apply: true          # Apply changes automatically
+
+# --- Optional: Set a default target environment ---
+# This is intended for local development to prevent users from accidentally applying plans to the prod environment.
+# It is a development only config and should NOT be committed to your git repo.
+# https://sqlmesh.readthedocs.io/en/stable/guides/configuration/#default-target-environment
+
+# Uncomment the following line to use a default target environment derived from the logged in user's name.
+# default_target_environment: dev_{{ user() }}
+
+# Example usage:
+# sqlmesh plan            # Automatically resolves to: sqlmesh plan dev_yourname
+# sqlmesh plan prod       # Specify `prod` to apply changes to production
+"""
+
+    return default_configs[template] + (flow_cli_mode if cli_mode == InitCliMode.FLOW else "")
+
+
+@dataclass
+class ExampleObjects:
+    schema_name: str
+    full_model_name: str
```

Comment:
I'd suggest to make it more generic without coupling this with a specific project structure:
```
models: t.Dict[str, str]
seeds: t.Dict[str, str]
...
```
with key being the name and the value being definition. For seeds the value could be tuple